atlassian_jwt_auth/contrib/django/middleware.py

```diff
@@ -53,3 +66,44 @@ def process_view(self, request, view_func, view_args, view_kwargs):
             request.META['HTTP_AUTHORIZATION'] = asap_auth
         if orig_auth is not None:
             request.META[self.xauth] = orig_auth
+
+
+class ASAPMiddleware(ASAPForwardedMiddleware):
+    """Enable ASAP for Django applications.
+
+    To use proxied credentials, this must come before any authentication
+    middleware.
+    """
+
+    def __init__(self, get_response=None):
+        with warnings.catch_warnings():
```

Comment:
I disagree with the way that `ASAPForwardedMiddleware` is deprecated but then the warning is captured here. Others who have extended `ASAPForwardedMiddleware` should not be forced to do the same.

I guess what I am saying is that I don't feel we should deprecate `ASAPForwardedMiddleware` at this point in time or if we want to do so we shouldn't be emitting a warning in the __init__(...) of `ASAPForwardedMiddleware`.

Reply:
> Others who have extended ASAPForwardedMiddleware should not be forced to do the same.

My plan was to either eliminate `ASAPForwardedMiddleware` or mark it private in a future PR, and have anybody using/extending it get a deprecation warning so they could plan accordingly.

> I guess what I am saying is that I don't feel we should deprecate ASAPForwardedMiddleware at this point in time [...]

I'd rather deprecate it early than risk having additional people rely on it.

> [I]f we want to do so we shouldn't be emitting a warning in the init(...) [...]

I prefer the warning, but only by a little bit. I'm strongly in favor of *some* form of deprecation; would dropping the warning and leaving the deprecation in the doc string be sufficient? Or is the warning just in the wrong place?

Reply:
Yep. That's fine my disagreement is mainly with the issued `warning`.